File: InvestmentPortfolioTracker/src/InvestmentPortfolioTracker.Core/Model/TransactionAggregate/Enums/TransactionType.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace InvestmentPortfolioTracker.Core;

/// <summary>
/// Represents the type of investment transaction.
/// </summary>
public enum TransactionType
{
    /// <summary>
    /// Buy transaction (purchase of securities).
    /// </summary>
    Buy = 0,

    /// <summary>
    /// Sell transaction (sale of securities).
    /// </summary>
    Sell = 1,

    /// <summary>
    /// Dividend received.
    /// </summary>
    Dividend = 2,

    /// <summary>
    /// Interest received.
    /// </summary>
    Interest = 3,

    /// <summary>
    /// Deposit (cash contribution).
    /// </summary>
    Deposit = 4,

    /// <summary>
    /// Withdrawal (cash distribution).
    /// </summary>
    Withdrawal = 5,

    /// <summary>
    /// Transfer between accounts.
    /// </summary>
    Transfer = 6,

    /// <summary>
    /// Fee or expense.
    /// </summary>
    Fee = 7,
}

File: InvestmentPortfolioTracker/src/InvestmentPortfolioTracker.Web/projects/api/src/lib/models/transaction-type.ts
export enum TransactionType {
  Buy = 0,
  Sell = 1,
  Dividend = 2,
  Interest = 3,
  Deposit = 4,
  Withdrawal = 5,
  Transfer = 6,
  Fee = 7
}

export const TRANSACTION_TYPE_LABELS: Record<TransactionType, string> = {
  [TransactionType.Buy]: 'Buy',
  [TransactionType.Sell]: 'Sell',
  [TransactionType.Dividend]: 'Dividend',
  [TransactionType.Interest]: 'Interest',
  [TransactionType.Deposit]: 'Deposit',
  [TransactionType.Withdrawal]: 'Withdrawal',
  [TransactionType.Transfer]: 'Transfer',
  [TransactionType.Fee]: 'Fee'
};
